ccl/changefeedccl: TestFetchChangefeedUsageBytesE2E failed

ccl/changefeedccl.TestFetchChangefeedUsageBytesE2E [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16588455?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16588455?buildTab=artifacts#/) on release-24.1 @ [7a4454ca9daa072a2bd0dc733745ae3189c282db](https://github.com/cockroachdb/cockroach/commits/7a4454ca9daa072a2bd0dc733745ae3189c282db):


```
=== RUN   TestFetchChangefeedUsageBytesE2E
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/f60805e43216da1104f4d49592543dce/logTestFetchChangefeedUsageBytesE2E3422210891
    test_log_scope.go:81: use -show-logs to present logs inline
    conditional_wrap.go:190: 
        pkg/ccl/changefeedccl_test/pkg/ccl/changefeedccl/fetch_table_bytes_test.go:430: (newUsageFixtureWithMockTSS)
        	NOTICE: .ExecutorConfig() called via implicit interface ApplicationLayerInterface;
        HINT: consider using .ApplicationLayer().ExecutorConfig() instead.
    conditional_wrap.go:194: TIP: consider replacing the test server initialization from:
            ts, ... := serverutils.StartServer(t, ...)
            defer ts.Stopper().Stop(...)
        to:
            srv, ... := serverutils.StartServer(t, ...)
            defer srv.Stopper().Stop(...)
            ts := srv.ApplicationLayer()
        
        See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
    fetch_table_bytes_test.go:227: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl_test/pkg/ccl/changefeedccl/fetch_table_bytes_test.go:227
        	Error:      	Should be zero, but was 1
        	Test:       	TestFetchChangefeedUsageBytesE2E
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f60805e43216da1104f4d49592543dce/logTestFetchChangefeedUsageBytesE2E3422210891
--- FAIL: TestFetchChangefeedUsageBytesE2E (3.16s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFetchChangefeedUsageBytesE2E.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41576

--- a/pkg/server/status.go
+++ b/pkg/server/status.go
@@ -3248,7 +3248,7 @@ func iterateNodes[Client, Result any](
 				responseFn(res.nodeID, res.response)
 			}
 		case <-ctx.Done():
-			resultErr = errors.Errorf("request of %s canceled before completion", errorCtx)
+			resultErr = errors.Wrapf(ctx.Err(), "request of %s canceled before completion", errorCtx)
 		}
 		numNodes--
 	}

